packages/xstate-store/test/store.test.ts

```diff
@@ -339,6 +339,43 @@ it('emitted events occur after the snapshot is updated', () => {
   store.send({ type: 'inc' });
 });
 
+it('events can be emitted with no payload', () => {
+  const spy = jest.fn();
+
+  const store = createStore({
+    emits: {
+      incremented: () => {},
+      decremented: () => {},
+      expectsPayload: (_: { payload: string }) => {}
```

Comment:
what about `_: { payload?: string }`? what should be accepted arguments for that?

Reply:
Ah let me add a test for that

Reply:
Just added a test:

```ts
  createStore({
    emits: {
      optionalPayload: (_: { payload?: string }) => {}
    },
    context: {},
    on: {
      inc: (_ctx, _ev, enq) => {
        enq.emit
          // @ts-expect-error
          .optionalPayload();

        enq.emit.optionalPayload({ payload: 'hello' });

        enq.emit.optionalPayload({});
      }
    }
  });
  ```